### PR TITLE
feat(samples): demonstrate real-time conversation translation

### PR DESCRIPTION
**Pull request by telpirion**
This PR improves the Media Translation microphone sample so that it is more aware of "state." When the API returns the SpeechEventType.END_OF_SINGLE_UTTERANCE event, the stream is closed and the user can initiate another call to the API.

The following region tags are affected:
media_translation_translate_from_mic

### CODE REVIEW

**Comment by telpirion** [comment]
I've set the ignore block on line 54 (call to `process.exit()`) to allow the user to immediately exit the sample.

**Comment by telpirion** [comment]
I added `eslint-disable-line no-process-exit` to samples/translate_from_mic.js because the call to `process.exit()` allows the user to exit the program gracefully. There is no error being caught here, and thus no reason to throw.

File: samples/translate_from_mic.js [hunk 1]
```diff
@@ -120,13 +141,13 @@ function main(encoding, sampleRateHertz, sourceLanguage, targetLanguage) {
           audioContent: chunk.toString('base64'),
         };
         stream.write(request);
+      })
+      .on('close', () => {
+        doTranslationLoop();
```
Comment (bcoe) [inline samples/translate_from_mic.js:146]: what is the point of this recursion, is it allowing someone to say, and translate, a few words in a row?

Comment (telpirion) [reply samples/translate_from_mic.js:146]: Something like that. It's to simulate one side of a conversation. The user can say something into their microphone, see it translated, (have the stream close), and then initiate another translation session.

This approximates some of the use cases that customers are looking for (real-time conversation translation).

**Comment by telpirion** [comment]
Thank you so much! 

Per our previous conversations, we should def. chat sometime to think through how we can improve how this sample handles streams.

**Pull request merged**